protos/sigstore_common.proto

```diff
@@ -0,0 +1,103 @@
+// See https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.180-4.pdf for more
+// details.
+// UNSPECIFIED SHOULD not be used, primary reason for inclusion is to force
+// any proto JSON serialization to emit the used hash algorithm, as default
+// option is to *omit* the default value of an emum (which is the first
+// value, represented by '0'.
+enum HashAlgorithm {
+        HASH_ALGORITHM_UNSPECIFIED = 0;
+        SHA2_256 = 1;
+        SHA2_512 = 2;
+}
+
+// HashOutput captures a digest of a 'message' (generic octet sequence)
+// and the corresponding hash algorithm used.
+message HashOutput {
+        HashAlgorithm algorithm = 1;
+        // This is the raw octets of the message digest as computed by
+        // the hash algorithm.
+        bytes digest = 2;
+}
+
+// MessageSignature stores the computed signature over a message.
+message MessageSignature {
+        // Message digest can be used to identify the artifact.
+        HashOutput message_digest = 1;
+        // The raw bytes as returned from the signature algorithm.
+        // The signature algorithm (and so the format of the signature bytes)
+        // are determined by the contents of the 'verification_material',
+        // either a key-pair or a certificate. If using a certificate, the
+        // certificate contains the required information on the signature
+        // algorithm.
+        // When using a key pair, the algorithm MUST be part of the public
+        // key, which MUST be communicated out-of-band.
+        bytes signature = 2;
+}
+
+// This message holds a RFC 3161 timestamp.
+message RFC3161SignedTimestamp {
+        // Signed timestamp is the DER encoded TimeStampResponse.
+        // See https://www.rfc-editor.org/rfc/rfc3161.html#section-2.4.2
+        bytes signed_timestamp = 1;
+}
+
+
+// PublicKeyIdentifier can be used to identify an (out of band) delivered
+// key, to verify a signature.
+message PublicKeyIdentifier {
+        // Optional unauthenticated hint on which key to use.
+        // The format of the hint must be agreed upon out of band by the
+        // signer and the verifiers, and so is not subject to this
+        // specification.
+        string hint = 1;
+}
+
+message X509Certificate {
+        // DER-encoded X.509 certificate.
+        bytes raw_bytes = 1;
+}
+
+// A chain of X.509 certificates.
+message X509CertificateChain {
+        // The chain of certificates, with indices 0 to n.
+        // The first certificate in the array must be the leaf
+        // certificate used for signing. Any intermediate certificates
+        // must be stored as offset 1 to n-1, and the root certificate at
+        // position n.
+        repeated X509Certificate certificates = 1;
+}
+
+// VerificationMaterial captures details on the materials used to verify
+// signatures.
+message VerificationMaterial {
+        oneof content {
+                PublicKeyIdentifier public_key = 1;
```

Comment:
I would say we are covered, as the `public_key` is just a hint, it could be used to identify what PGP key in your key ring. So it's still the verifiers responsibility to prepare all the key material upfront, prior to the verification starts. We can hopefully start to work on this more this afternoon.

Reply:
But why are we allowing a cert chain to be specified here but not a pgp key? Sorry, perhaps I need more caffeine this morning

Reply:
You mean that based on transitive/web of trust I can trust an unknown key if it's signed by someone I trust?

Even for the cert chain, the root cert is not to be trusted unless it is also provided by the verifier.

Reply:
Are you ok that we add an issue for adding support for PGP keys and address it post first merge/version? As that would be a new option for the `VerificationMaterial` it's a change we can add that is backward compatible.

Reply:
Since this is a `oneof` I think that’s fine